prov/util/src/util_buf.c

```diff
@@ -55,14 +55,11 @@ static void ofi_bufpool_set_region_size(struct ofi_bufpool *pool)
 	}
 
 	hp_size = ofi_get_hugepage_size();
-	if ((pool->attr.chunk_cnt + 1) * pool->entry_size >= hp_size) {
-		pool->attr.flags |= OFI_BUFPOOL_MMAPPED;
+	if (pool->attr.flags & OFI_BUFPOOL_MMAPPED)
```

Comment:
Yes, will add some reasoning to the commit message. When huge page support was added to the bufpools, it was getting used by default for all providers. This allows providers to decide which pools should use them and opt-in explicitly (this will be useful for cases like [this one](https://github.com/ofiwg/libfabric/issues/4969)). This could still use some work to fallback to regular pages when huge-pages are depleted, but that patch is yet to be written.